authorship/data.py

```diff
@@ -57,13 +55,12 @@ def __call__(self, texts):
 
 
 class RearrangeSamples:
-    def __init__(self, n, wordlist):
-        self.n = n
+    def __init__(self, wordlist):
         self.wordlist = wordlist
 
     def __call__(self, items):
-        for i in range(len(items) // self.n):
-            yield [word for word in items[i*self.n: (i+1)*self.n] if word in self.wordlist]
+        yield [word for word in items if word in self.wordlist]
```

Comment:
If it is just to filter words, this class could be replaced by something simpler.

Reply:
I removed both classes (RearrangeSamples and ExtractWords) as they made more sense in the previous setting